File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    answer = ''
    
    per = ['R', 'T', 'C', 'F', 'J', 'M', 'A', 'N']  # 성격 유형
    per_cnt = [0, 0, 0, 0, 0, 0, 0, 0]              # 성격 점수 
    
    ag_dag = [0, 3, 2, 1, 0, 1, 2, 3]               # 동의 비동의
    for sur, cho in zip(survey, choices):
        per1, per2 = sur[0], sur[1]                 # 성격 유형 자르기
        if cho < 4:                                 # 질문에서 비동의를 체크했을 때
            idx = per.index(per1)                   
            per_cnt[idx] += ag_dag[cho]
        else:                                       # 질문에서 동의를 체크했을 때
            idx = per.index(per2)
            per_cnt[idx] += ag_dag[cho]
    
    for i in range(0, len(per_cnt), 2):
        if per_cnt[i] == per_cnt[i + 1]:            # 점수가 같으면 사전순 앞에 유형
            answer += per[i]
        elif per_cnt[i] < per_cnt[i + 1]:           # 점수가 높은 성격 찾기
            answer += per[i + 1]
        else:
            answer += per[i]

    return answer

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    var answer = '';
    const personality = ['R', 'T', 'C', 'F', 'J', 'M', 'A', 'N'];        // 성격 유형
    const personality_score = [0, 0, 0, 0, 0, 0, 0, 0];                  // 유형 별 점수
    const score = [3, 2, 1, 0, 1, 2, 3];                                 // 질문에 따른 점수 배치
  
    for (let i = 0; i < survey.length; i++) {
        const [sur1, sur2] = survey[i].split('');
        let idx = 0;
        if (choices[i] >= 5) idx = personality.indexOf(sur2);   // 5 이상(약간 동의)부터 다음 유형에 점수를 줘야하므로 조건 처리
        else idx = personality.indexOf(sur1);
        personality_score[idx] += score[choices[i] - 1];
    }
    
    for (let i = 0; i < personality_score.length; i += 2) {
        if (personality_score[i] >= personality_score[i + 1]) answer += personality[i];
        else answer += personality[i + 1];
    }
    
    return answer;
}